### PR TITLE
This concludes #73

### PR DESCRIPTION
**Pull request by osleonard**
- added java fmt
- bumped up sbt version to required version for sbt-java-formatter

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: examples/sensor-data-java/project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version = 1.2.8
+sbt.version = 1.3.0
```
Comment (olofwalker) [inline examples/sensor-data-java/project/build.properties]: We are keeping SBT at `1.2.8` in all projects for now.

Comment (RayRoestenburg) [reply examples/sensor-data-java/project/build.properties]: https://github.com/sbt/sbt-java-formatter/commit/48bb4c45d7b71a128652315d8f15b6ec9117510c @olofwalker

**Comment by agemooij** [comment]
@osleonard could you please double-check your Github email settings to see if we can fi the warning posted above about your commits not being linked to a recognized user? You probably committed with a git email address not linked to your Github account.

Here's how to fix it: https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: examples/sensor-data-java/project/build.properties (hunk 1 above)
Comment (agemooij) [reply examples/sensor-data-java/project/build.properties]: Seems like it is time to upgrade to 1.3.x. We will pick that up ASAP to unblock this PR

**Comment by osleonard** [comment]
> @osleonard could you please double-check your Github email settings to see if we can fix the warning posted above about your commits not being linked to a recognized user? You probably committed with a git email address not linked to your Github account.
> 
> Here's how to fix it: https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

Fixed :)

**Comment by agemooij** [comment]
@osleonard a few requests:

- I only see the java formatter applied to some of the examples but the core also has some Java files, in both main and test scopes. Could you please also add the Java formatter to the relevant (or all) core projects?
- Could you please upgrade to the latest version of sbt `1.3.x`, which at the time of writing this is `1.3.8`. the initial release has a number of bugs that prevented us from upgrading when it first came out.
- Could you perhaps upgrade all sbt builds in the project, including the core, to 1.3.8?

Thanks a lot for the help!

**Comment by osleonard** [comment]
> @osleonard a few requests:
> 
>     * I only see the java formatter applied to some of the examples but the core also has some Java files, in both main and test scopes. Could you please also add the Java formatter to the relevant (or all) core projects?
> 
>     * Could you please upgrade to the latest version of sbt `1.3.x`, which at the time of writing this is `1.3.8`. the initial release has a number of bugs that prevented us from upgrading when it first came out.
> 
>     * Could you perhaps upgrade all sbt builds in the project, including the core, to 1.3.8?
> 
> 
> Thanks a lot for the help!

Glad to work on the above

**Comment by skyluc** [comment]
I test `sensor-data-java` compiled with sbt 1.3.8 on a cluster.
I didn't see any problems.

**Comment by osleonard** [comment]
> I test `sensor-data-java` compiled with sbt 1.3.8 on a cluster.
> I didn't see any problems.

Bumped sbt version to 1.3.8 across project.

**Comment by RayRoestenburg** [comment]
@osleonard call-record-aggregator example also has some java code, could you add it there as well? And as @agemooij said it would be great if you also added it to core projects that have java code:
```
cloudflow-akka
cloudflow-akka-util
cloudflow-akka-testkit
cloudflow-akka-tests
cloudflow-flink-tests
```

**Comment by agemooij** [comment]
@osleonard FYI: the build is failing due to (as far as we can tell) a problem caused by sbt 1.3.x, e.g. it has caused some kind of classloader behavior change that is crashing the unit tests of the `call-record-aggregator` example app.

We will look into this internally but let us know if you have seen this before.

**Comment by osleonard** [comment]
> @osleonard FYI: the build is failing due to (as far as we can tell) a problem caused by sbt 1.3.x, e.g. it has caused some kind of classloader behavior change that is crashing the unit tests of the `call-record-aggregator` example app.
> 
> We will look into this internally but let us know if you have seen this before.

Taking a look would let you know what I find

**Comment by osleonard** [comment]
> > @osleonard FYI: the build is failing due to (as far as we can tell) a problem caused by sbt 1.3.x, e.g. it has caused some kind of classloader behavior change that is crashing the unit tests of the `call-record-aggregator` example app.
> > We will look into this internally but let us know if you have seen this before.
> 
> Taking a look would let you know what I find

Figured out the problem, sbt 1.3.8 is not compatible with spark 2.4.4, I updated to 2.4.5 published a local snapshot and tested. All test became green. Seen screenshot.
Screenshot 2020-02-19 at 20 01 12

**Comment by agemooij** [comment]
I figured out the root cause, which is that the default way sbt 1.3.x resolves dependencies is slightly different from how sbt 1.2.x did it, which leads to a different dependency classpath that causes an incompatible version of Netty to be on the classpath.

We already have an PR in progress to upgrade to Spark 2.4.5, so one option is to wait for that Pr to be merged and then rebase this Pr on top of that.

Another option would be to the temporarily fix this PR by disabling the new dependency resolver (`Global / useCoursier := false`), which also solves the problem, and then potentially revert that once the Spark 2.4.5 dependencies are merged and they test out.

I'll check today when we can merge the Spark 2.4.5 stuff and get back to you.

**Comment by osleonard** [comment]
> I figured out the root cause, which is that the default way sbt 1.3.x resolves dependencies is slightly different from how sbt 1.2.x did it, which leads to a different dependency classpath that causes an incompatible version of Netty to be on the classpath.
> 
> We already have an PR in progress to upgrade to Spark 2.4.5, so one option is to wait for that Pr to be merged and then rebase this Pr on top of that.
> 
> Another option would be to the temporarily fix this PR by disabling the new dependency resolver (`Global / useCoursier := false`), which also solves the problem, and then potentially revert that once the Spark 2.4.5 dependencies are merged and they test out.
> 
> I'll check today when we can merge the Spark 2.4.5 stuff and get back to you.

Think it would be better to wait for the 2.4.5 change rather than disabling the dependency resolver IMHO, but anything you decide is fine.

**Comment by agemooij** [comment]
@osleonard we just merged the Spark 2.4.5 changes to master. could you please try to rebase and see if that fixes the issue in the tests for the CRA example?

If not, we can still disable the Coursier resolver in sbt if we have to of course.

**Comment by osleonard** [comment]
> @osleonard we just merged the Spark 2.4.5 changes to master. could you please try to rebase and see if that fixes the issue in the tests for the CRA example?
> 
> If not, we can still disable the Coursier resolver in sbt if we have to of course.

Would rebase off the current master update the PR

**Comment by agemooij** [comment]
@osleonard yes, a rebase followed by a forced push would update the PR.

**Comment by osleonard** [comment]
@agemooij Done, believe is ready for merge

**Pull request merged**